### ISSUE TITLE
(C#) Bot is connecting to my Chat with my Streamer Account not the specific Bot-Account

### ISSUE THREAD

**Issue by Snickbrack** [TwitchLib/TwitchLib#1121]
I have a Streamer-Account and a separate Bot-Account just for the Twitch Chat-Bot which uses TwitchLib.

When I start my Bot it connects to my Stream-Chat. But with my Streamer-Account. And also sends Messages to the Chat as my Streamer-Account.

The Bot uses his own Twitch-Username and the OAuth-Token which I create with this Link (https://twitchapps.com/tokengen/) with the Client-ID for the Application (of the Bot-Twitch Account) and the required Rights. I connect to TwitchApps with my Streamer-Account to authorize the granting of the rights.

And the code for connection looks like this:

```
ConnectionCredentials credentials = new ConnectionCredentials(channelInformation.BotUserName, channelInformation.AccessToken);

client = new TwitchClient(customClient);
client.Initialize(credentials, channelInformation.ChannelName);

client.AddChatCommandIdentifier('!');

client.OnLog += Client_OnLog;
client.OnJoinedChannel += Client_OnJoinedChannel;
client.OnMessageReceived += Client_OnMessageReceived;
client.OnConnected += Client_OnConnected;
client.OnChatCommandReceived += Client_OnChatCommandReceived;
PubSub.Connect();
client.Connect();
```

And sending code looks like this:

`client.SendMessage(channelInformation.ChannelName, "Aktuell gibt es keine Giveaways Sadge");`

The Message in Chat looks like this:
![image](https://github.com/TwitchLib/TwitchLib/assets/28804584/0811bde0-7db2-41d0-9dee-fa3deef462c4)

But the Bot-Username is "Anwalt_von_Kage". And the Credentials inside the Bot also looks like this:
![image](https://github.com/TwitchLib/TwitchLib/assets/28804584/b11510e9-fd9e-4ff4-a6de-b86f4dc1af21)

And the Bot isn't visible in the Chatters-List in my Twitch-Chat and also not visible as online in the Moderation-Overview:
![image](https://github.com/TwitchLib/TwitchLib/assets/28804584/81e3bf78-2cfb-4572-acc3-61a63704b30a)

Can somebody tell me something about what is going on there?

**Comment by Mahsaap** [comment]
The client uses the token alone to determine the user being used to send a message. Doesn't matter what you put in the username field. 
Ensure the token in 
`ConnectionCredentials credentials = new ConnectionCredentials(channelInformation.BotUserName, channelInformation.AccessToken);`
was generated with the bot accounts dashboard if you want the bot to send the message.
if you generated this with a client ID from the streamer account, I believe it will still use the streamer since they own it.

**Comment by Snickbrack** [comment]
> The client uses the token alone to determine the user being used to send a message. Doesn't matter what you put in the username field. Ensure the token in `ConnectionCredentials credentials = new ConnectionCredentials(channelInformation.BotUserName, channelInformation.AccessToken);` was generated with the bot accounts dashboard if you want the bot to send the message. if you generated this with a client ID from the streamer account, I believe it will still use the streamer since they own it.

Hi and thanks for your reply. But my Bot uses the ClientID which I have got from dev.twitch.tv for the Bot-Account "Anwalt_von_Kage" which is my Bot-Account. And it worked for long time and then it "stopped" working as expected and switched to my Streamer-Account. But without any Change from my side...

**Comment by Snickbrack** [comment]
Also, I have just checked my logs and this is logged:

2023-02-04 22:00:29.6680|DEBUG|Twitch_Chat_Bot.Classes.Bot.Bot|04.02.2023 21:00:29: anwalt_von_kage - Received: :anwalt_von_kage!anwalt_von_kage@anwalt_von_kage.tmi.twitch.tv JOIN #bullkagex


And for yesterday it is:

2023-05-19 21:42:59.7975|DEBUG|Twitch_Chat_Bot.Classes.Bot.Bot|19.05.2023 19:42:59: anwalt_von_kage - Received: :bullkagex!bullkagex@bullkagex.tmi.twitch.tv JOIN #bullkagex


But for real the ClientID is the one for the Twitch Application for Anwalt_von_Kage and I authorize it with my Streamer-Account.

And in Code there were no changes made...

**Comment by Snickbrack** [comment]
Okay ehhmmm. wtefff..,.

I do need two OAuths now? One for Anwalt_von_Kage to control this one to be able to chat in my Streaming-Account and one that can Acces the Twitch-API for Bullkagex?

I dont needed this forever and I was able to acces both Data-"Endpoint" just with the Anwalt_von_Kage OAuth which I have authorized with my Bullkagex-Account through the [implicit grant flow](https://dev.twitch.tv/docs/v5/guides/authentication/) and this TokenGen-Website...

What the smack.........
